### PR TITLE
update noRevoke

### PR DESCRIPTION
**Pull request by vhvb1989**
fixes: #1102

### CODE REVIEW

File: sdk/core/azure-core/inc/azure/core/http/curl/curl.hpp
Comment (ahsonkhan) [inline sdk/core/azure-core/inc/azure/core/http/curl/curl.hpp:34]: nit: It's nice to have default values be the same as "0", so `false` for bool. This is quite common in .NET:
https://docs.microsoft.com/en-us/dotnet/csharp/language-reference/builtin-types/default-values

In C++, default values of bool is false, as well.

So, consider naming it Enable rather than Disable, and negate it.

```suggestion
    bool EnableCertificateRevocationListCheck = false;
```

Comment (vhvb1989) [reply sdk/core/azure-core/inc/azure/core/http/curl/curl.hpp:34]: The default value from libcurl here is `false`, We mention that in the comments. Then we mention how we are explicitly changing this for the SDK.

Comment (ahsonkhan) [reply sdk/core/azure-core/inc/azure/core/http/curl/curl.hpp:34]: >The default value from libcurl here is `false`, We mention that in the comments. Then we mention how we are explicitly changing this for the SDK.

I am aware of that :)

As a customer of the SDK, it doesn't matter what the libcurl defaults are. I recommend decoupling implementation detail from the public API surface. The doc comment as-is continues to hold true, even with my suggestion.

Comment (vhvb1989) [reply sdk/core/azure-core/inc/azure/core/http/curl/curl.hpp:34]: I will play the old and hated classic quote:  "this PR is following the requested change in the issue", please approve and create a new issue if the requirement has changed.

Let's keep requirements clear and in the issue only and avoid updating the expectation on PRs.

Comment (ahsonkhan) [reply sdk/core/azure-core/inc/azure/core/http/curl/curl.hpp:34]: > "this PR is following the requested change in the issue"

I quoted the issue above. It explicitly indicated that we would want to flip disable -> enable based on the default value change (at the time, the default value was `true`). The primary goal of the issue was to avoid double negative, and we are still somewhat in that came.

I don't see this as a change of requirements or going against the issue at all.

That said, I'll leave it up to you to decide to take the suggestion, if you feel strongly about keeping it `disable`, and leaving the default value as `false`.

**Review by ahsonkhan — APPROVED** [review]
I recommend the name `EnableCertificateRevocationListCheck` or `CheckCertificateRevocationList`, to get idiomatic C++ bool default value, but will leave it up to you.

File: sdk/core/azure-core/src/http/curl/curl.cpp
Comment (ahsonkhan) [inline sdk/core/azure-core/src/http/curl/curl.cpp:1205]:
```suggestion
  if (!options.SSLOptions.EnableCertificateRevocationListCheck)
```

Comment (vhvb1989) [reply sdk/core/azure-core/src/http/curl/curl.cpp:1205]: thanks. I was just debugging this 👍

**Review by ahsonkhan — APPROVED** [review]

File: sdk/core/azure-core/src/http/curl/curl.cpp
Comment (ahsonkhan) [inline sdk/core/azure-core/src/http/curl/curl.cpp:1080]: Was this change necessary for correctness? I would imagine if someone set `EnableCRL = true`, that would mean you append  "1" to the key, but if CRL checking is off, you'd append "0".  Why negate it here?

Comment (vhvb1989) [reply sdk/core/azure-core/src/http/curl/curl.cpp:1080]: yes, it was either changing all unit tests from 1 to 0 or change the sentinel here.

**Pull request merged**
